test/temporary_directory_fixture.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TEMPORARY_DIRECTORY_FIXTURE_HPP_
+#define TEMPORARY_DIRECTORY_FIXTURE_HPP_
+
+#include <gmock/gmock.h>
+
+#include <string>
+
+#ifdef _WIN32
+# include <direct.h>
+# include <Windows.h>
+#else
+# include <unistd.h>
+# include <sys/types.h>
+# include <dirent.h>
+#endif
+
+
+class TemporaryDirectoryFixture : public ::testing::Test
+{
+public:
+  TemporaryDirectoryFixture()
```

Comment:
1/ this code needs to be re-architectured into small functions, the functions which are generic enough should be moved to rcpputils itself, and should have tests. They should also be moved to their own PRs.
2/ fixtures ctors should not do anything if it can be avoided, those complex operations should be moved to `SetUp` and `TearDown` instead.
3/ This test should be run with `--gtest_repeat` to ensure the creation/destruction logic is working properly in presence of flags.